controller/src/main/java/org/jboss/as/controller/persistence/ConfigurationFile.java

```diff
@@ -369,7 +370,9 @@ private File determineMainFile(final String rawName, final String name) {
         if (mainName != null) {
             try {
                 final File mainFile = new File(configurationDir, name != null ? name : mainName);
-                if (mainFile.getCanonicalPath().startsWith(configurationDir.getCanonicalPath())) {
+                Path file = Paths.get(mainFile.getPath());
```

Comment:
Thanks @parsharma , just a suggestion, you have already a File object, if you need a Path to check the symbolic link on the right side of || you could just convert a File into a path, that will avoid creating this object if it is not necessary
```
if (mainFile.getCanonicalPath().startsWith(configurationDir.getCanonicalPath()) || Files.isSymbolicLink(mainFile.toPath))
```

A test case for this is more than welcome

Reply:
Done